ebpf/common.go

```diff
@@ -3,16 +3,95 @@ package ebpf
 import (
 	"encoding/binary"
 	"errors"
+	"fmt"
 	"unsafe"
 )
 
+var (
+	// Feature versions sourced from: https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md
+	// Minimum kernel version -> max(3.15 - eBPF,
+	//                               3.18 - tables/maps,
+	//                               4.1 - kprobes,
+	//                               4.3 - perf events)
+	// 	                      -> 4.3
+	minRequiredKernelCode = linuxKernelVersionCode(4, 3, 0)
+)
+
 var (
 	// ErrNotImplemented will be returned on non-linux environments like Windows and Mac OSX
 	ErrNotImplemented = errors.New("BPF-based network tracing not implemented on non-linux systems")
 
 	nativeEndian binary.ByteOrder
 )
 
+func kernelCodeToString(code uint32) string {
+	// Kernel "a.b.c", the version number will be (a<<16 + b<<8 + c)
+	a, b, c := code>>16, code>>8&0xf, code&0xf
+	return fmt.Sprintf("%d.%d.%d", a, b, c)
+}
+
+func stringToKernelCode(str string) uint32 {
+	var a, b, c uint32
+	fmt.Sscanf(str, "%d.%d.%d", &a, &b, &c)
+	return linuxKernelVersionCode(a, b, c)
+}
+
+// KERNEL_VERSION(a,b,c) = (a << 16) + (b << 8) + (c)
+// Per https://github.com/torvalds/linux/blob/master/Makefile#L1187
+func linuxKernelVersionCode(major, minor, patch uint32) uint32 {
+	return (major << 16) + (minor << 8) + patch
+}
+
+var ubuntu44119Error = "got ubuntu kernel %s with known bug, see: https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1763454"
+var defaultExclusionList = map[uint32]string{
+	linuxKernelVersionCode(4, 4, 119): ubuntu44119Error,
```

Comment:
This doesn't actually confirm that it's ubuntu though. Just the linux kernel itself. Isn't that an issue?

Reply:
Indeed the issue seems to be only on Ubuntu, I will add something to check the OS name

Reply:
Might be worth looking at `bpflib.CurrentKernelVersion()`. They do some semi-complicated parsing of the `uname` output